frontend/pages/hosts/ManageHostsPage/components/LabelFilterSelect/LabelFilterSelect.tsx

```diff
@@ -39,23 +43,25 @@ const formatOptionLabel = (data: ILabel | IEmptyOption) => {
   const isLabel = "display_text" in data;
   const isPlatform = isLabel && data.type === "platform";
 
-  let labelText = isLabel ? data.display_text : data.label;
+  let displayText = isLabel ? data.display_text : data.label;
 
   // the display names for platform options are slightly different then the display_text
   // property, so we get the correct display name here
   if (isLabel && isPlatform) {
-    labelText = PLATFORM_LABEL_DISPLAY_NAMES[data.display_text];
+    if (isPlatformLabelNameFromAPI(data.display_text)) {
+      displayText = PLATFORM_LABEL_DISPLAY_NAMES[data.display_text];
+    }
   }
 
   return (
     <div className="option-label">
-      {isPlatform && (
+      {isLabel && hasPlatformTypeIcon(data.display_text) && (
```

Comment:
This was causing a page crash in certain direct navigation scenarios as described above.